### PR TITLE
feat(php): add installed.json file support

### LINKED ISSUE #5057: feat(php): add installed.json files support
## Description
Add support `path_to_app/vendor/composer/installed.json` files.

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add `installed.json` file support.
Supported modes - `image` and `rootfs`.

**Scanning `composer.lock` files is only possible in `fs` and `repo` modes now**

Example of work:
```
➜  ./trivy rootfs ./installed.json 
2023-08-29T13:24:06.376+0600    INFO    Vulnerability scanning is enabled
2023-08-29T13:24:06.376+0600    INFO    Secret scanning is enabled
2023-08-29T13:24:06.376+0600    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-08-29T13:24:06.376+0600    INFO    Please see also https://aquasecurity.github.io/trivy/dev/docs/scanner/secret/#recommendation for faster secret detection
2023-08-29T13:24:06.387+0600    INFO    Number of language-specific files: 1
2023-08-29T13:24:06.387+0600    INFO    Detecting composer-installed vulnerabilities...

installed.json (composer-installed)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 2, HIGH: 0, CRITICAL: 0)

┌─────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬────────────────────────────────────────────────────┐
│     Library     │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                       Title                        │
├─────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼────────────────────────────────────────────────────┤
│ guzzlehttp/psr7 │ CVE-2022-24775 │ MEDIUM   │ fixed  │ 1.8.3             │ 1.8.4, 2.1.1  │ Improper Input Validation in guzzlehttp/psr7       │
│                 │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2022-24775         │
│                 ├────────────────┤          │        │                   ├───────────────┼────────────────────────────────────────────────────┤
│                 │ CVE-2023-29197 │          │        │                   │ 1.9.1, 2.4.5  │ Improper header name validation in guzzlehttp/psr7 │
│                 │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2023-29197         │
└─────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴────────────────────────────────────────────────────┘

```

## Related Issues:
- Close #5057 

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by AnaisUrlichs — APPROVED** [review]

File: pkg/fanal/analyzer/const.go [hunk 1]
```diff
@@ -49,7 +49,8 @@ const (
 	TypeCargo      Type = "cargo"
 
 	// PHP
-	TypeComposer Type = "composer"
+	TypeComposer          Type = "composer"
+	TypeComposerInstalled Type = "composer-installed"
```
Comment (knqyf263) [inline pkg/fanal/analyzer/const.go:53]: How about `composer-vendor`?

```suggestion
	TypeComposerInstalled Type = "composer-vendor"
```

File: pkg/fanal/analyzer/language/php/installed/installed.go
Comment (knqyf263) [inline pkg/fanal/analyzer/language/php/installed/installed.go:1]: I think this file should be located under `php/composer` as `installed.json` is a composer-specific file.

**Review by knqyf263 — COMMENTED** [review]
Sorry to be late

File: pkg/fanal/analyzer/const.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/const.go]: `intalled.json` file placed in `path_to_app/vendor/...` dir.
So it make sense.
Updated.

File: pkg/fanal/analyzer/language/php/installed/installed.go
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/language/php/installed/installed.go:1]: moved

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**